File: src/co/soundwave/repository/playlist/PlaylistDAO.kt
package co.soundwave.repository.playlist

import co.soundwave.models.Playlist
import co.soundwave.repository.BaseDAO
import java.sql.ResultSet

class PlaylistDAO : BaseDAO<PlaylistSQL, Playlist>() {
    override val sql: PlaylistSQL by lazy { PlaylistSQL() }
    
    fun getInsertQuery(obj: Playlist, userId: Int): String =
        sql.insertPlaylist(obj.name, userId)
    
    fun queryByUserId(userId: Int): ResultSet? =
        queryExecutor(sql.queryByUserId(userId))
    
    fun addSongToPlaylist(playlistId: Int, songId: Int): Boolean =
        insert("insert into playlist_has_song values ('$playlistId', '$songId')")
}

File: repository/playlist/PlaylistDAO.py
from models import Playlist as pl
from repository import BaseDAO as bdao
from repository.playlist import PlaylistSQL as plsql


class PlaylistDAO(bdao.BaseDAO):
    def __init__(self):
        super().__init__()
        self.sql = plsql.PlaylistSQL()

    def get_insert_query(self, pla: pl.Playlist, user_id: int) -> str:
        return self.sql.insert_playlist(pla.name, user_id)

    def query_by_user_id(self, user_id: int) -> []:
        return self.query_executor(self.sql.query_by_user_id(user_id))

    def add_song_to_playlist(self, playlist_id: int, song_id: int) -> bool:
        return self.insert("insert into playlist_has_song values(%s, %s);" % (playlist_id, song_id))
